taskvine/src/manager/vine_manager.c

```diff
@@ -3259,13 +3284,14 @@ static int send_one_task(struct vine_manager *q)
 				continue;
 		}
 
-		// Otherwise, remove it from the ready list and start it:
-		list_pop_tail(q->ready_list);
+		// Otherwise, remove it from the ready queue and start it:
+		priority_queue_remove(q->ready_tasks, t);
+
 		commit_task_to_worker(q, w, t);
 		return 1;
-	}
+	} while ((priority_queue_update_priority(q->ready_tasks, t, --t->priority)) && (t = priority_queue_scheduling_next(q->ready_tasks)));
```

Comment:
I don't think the scheduler should be changing the priority of tasks.
That information comes from the outside, and the scheduler should respect it.